drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -146,6 +162,23 @@ local function device_init(driver, device)
     device:set_endpoint_to_component_fn(endpoint_to_component)
     device:set_find_child(find_child)
     device:subscribe()
+
+    if not device:get_field(BOUNDS_CHECKED) then
+      local limit_read
```

Comment:
You could just initialize the `limit_read` variable up here
```
limit_read = im.InteractionRequest(im.InteractionRequest.RequestType.READ, {})
```

And then merge the Color Temperature and Level limit reads into one read request you send at the end of the function instead of two separate reads.

The `BOUNDS_CHECKED` flag you have added here will properly address [Tom's comment ](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1220#discussion_r1513389233) about doing this once for each device.

Reply:
I initially had it that way and it caused failures in other tests within the matter-switch driver, because some of the mock devices in those tests support neither the colorTemperature nor the switchLevel capability. This caused it to send a blank InteractionRequest in those tests. I decided to make the changes in c1d9f8213eb383876938c93f5b7e0b657a96ea42 to ensure that it would only send read requests if the capabilities are actually supported.

Reply:
Can we do a check for emptiness of the request and them omit sending it if empty?

Reply:
Updated code to check for emptiness of the request as suggested in 775564f06809491f7f3c5359f887bf7864edad48.